### PR TITLE
Added initialisation for simple and mongoose stores

### PR DESCRIPTION
**Pull request by nialldonnellyfh**
# Motivation

When the cloud app starts, we should use `fh-wfm-mongoose-store` if the mongo connection url is available.

If there is no mongo url, we should default to the `fh-wfm-simple-store` that can use the $fh.db API to store data.

### CODE REVIEW

File: lib/storage/mongoUtil.js [hunk 1]
```diff
@@ -2,14 +2,18 @@ var $fh = require('fh-mbaas-api');
...
+ *
+ * @param {function} cb
  */
 function getMongoURL(cb) {
   // Check if direct mongo connection url exist (available after upgrading db) as mongoStore requires
   // direct url and will throw an error when trying to connect.
   $fh.db({
     act: 'connectionString'
   }, function(err, connectionString) {
+    console.log("err, connectionString", err, connectionString);
```
Comment (wtrocki) [inline lib/storage/mongoUtil.js]: Let's remove this

File: lib/storage/mongoose-store.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+function disconnect() {
+  return store.disconnect();
+}
+
+/**
+ *
+ * @param dataSetId
+ * @returns {*}
+ */
+function getCollectionDal(dataSetId) {
```
Comment (wtrocki) [inline lib/storage/mongoose-store.js]: We can rename it to getConnection (DAL shortcut is not obvious)

File: lib/storage/mongoose-store.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  return store.getDAL(dataSetId);
+}
+
+module.exports = {
+  connect: connect,
+  disconnect: disconnect,
+  getCollectionDal: getCollectionDal
```
Comment (wtrocki) [inline lib/storage/mongoose-store.js]: Same as above. DAL in this case may be DAO, but best to not mention this patterns and simply use storage.
getCollectionStore?

File: lib/storage/mongoUtil.js (hunk 1 above)
Comment (nialldonnellyfh) [inline lib/storage/mongoUtil.js]: Will do, thanks.

File: lib/storage/mongoose-store.js (hunk 3 above)
Comment (nialldonnellyfh) [inline lib/storage/mongoose-store.js]: Yes. Makes sense

File: lib/app.js [hunk 4]
```diff
@@ -66,6 +66,18 @@ app.use(require('fh-wfm-user/lib/cloud/middleware/validateSession')(mediator, mb
...
 
+/**
+ *
+ * We are using mongoose models that have a `updatedAt` field.
+ *
+ * This can be used to detect if the record has changed.
+ *
+ * @param record
+ */
+syncOptions.hashFunction = function generateHash(record) {
```
Comment (wtrocki) [inline lib/app.js]: Going to move this only for mongoose as this seems to be breaking simple store implementation

File: package.json [hunk 5]
```diff
@@ -30,8 +30,8 @@
     "fh-wfm-user": "0.6.0",
     "fh-wfm-workflow": "0.2.2",
     "fh-wfm-workorder": "0.2.2",
-    "fh-wfm-sync": "0.1.2",
-    "raincatcher-mongoose-store": "0.1.0-pre.1",
+    "fh-wfm-sync": "0.2.0-pre.1",
+    "raincatcher-mongoose-store": "0.1.0-pre.2",
```
Comment (wtrocki) [inline package.json]: Going to move package name to fh-wfm-mongoose-store to be consistent and remove previous one from npm.

**Pull request merged**